File: sample/src/main/java/vipsffm/SampleHelper.java
package vipsffm;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class SampleHelper {

    public static void validate(
        Path path,
        long expectedMinSizeKb,
        long expectedMaxSizeKb
    ) throws IOException {
        if (!Files.exists(path)) {
            throw new RuntimeException("wrote thumbnail, but failed to find at \"" + path + "\"");
        }

        var fileSizeKb = Files.size(path) / 1000;
        if (fileSizeKb < expectedMinSizeKb || fileSizeKb > expectedMaxSizeKb) {
            throw new RuntimeException("unexpected resulting thumbnail size " + fileSizeKb + "kb");
        }

        var bytes = Files.readAllBytes(path);
        var allZeroes = true;
        for (var b : bytes) {
            if (b != 0) {
                allZeroes = false;
                break;
            }
        }
        if (allZeroes) {
            throw new RuntimeException("file all zeroes");
        }
    }
}

File: sample/src/main/kotlin/vipsffm/SampleHelper.kt
package vipsffm

import app.photofox.vipsffm.jextract.VipsRaw
import java.nio.file.Files
import java.nio.file.Path

object SampleHelper {

    fun runCatching(vipsFunction: () -> Int): Result<Unit> {
        val result = vipsFunction()
        if (result != 0) {
            val error = VipsRaw.vips_error_buffer().getString(0)
            return Result.failure(
                RuntimeException("failed to run vips command: $error")
            )
        }
        return Result.success(Unit)
    }

    fun validate(
        path: Path,
        expectedSizeBoundsKb: LongRange
    ): Result<Unit> {
        if (!Files.exists(path)) {
            return Result.failure(
                RuntimeException("wrote thumbnail, but failed to find at \"$path\"")
            )
        }

        val fileSizeKb = Files.size(path) / 1000
        if (!expectedSizeBoundsKb.contains(fileSizeKb)) {
            return Result.failure(
                RuntimeException("unexpected resulting thumbnail size $fileSizeKb")
            )
        }

        return Result.success(Unit)
    }
}